### ISSUE TITLE
No tensorboard logs during training

### ISSUE THREAD

**Issue by waldiwunder** [tencent-ailab/IP-Adapter#409]
Hi! Thank you for your interesting work. I would like to train the adapter on a custom dataset.

Currently, not logs are saved during the training. I checked the issue #287 and the link to https://github.com/huggingface/diffusers/tree/main/examples/text_to_image, but it did not help.

I want to use the default logging with tensorboard and call

`accelerate launch --mixed_precision "fp16" tutorial_train.py --pretrained_model_name_or_path="runwayml/stable-diffusion-v1-5" --image_encoder_path="models/image_encoder/" --data_json_file="custom_dataset/file.json" --data_root_path="path-to-images" --resolution=512 --train_batch_size=1 --dataloader_num_workers=1 --learning_rate=1e-04 --weight_decay=0.04 --output_dir="training_log/" --save_steps=10000 --report_to="tensorboard"`

I have only 1 GPU, so I do not use --multi-gpu

From the code I would expect that a directory training_log/runs/**CURRENT_DATETIME_HOSTNAME*** would be created with an event file, but nothing happens. I tried also with wandb (installing wandb and logging to my account), but also here nothing happens.

Could you give me an advice about what I am doing wrong? Thank you very much!

**Comment by flankechen** [comment]
I did it with wandb, current training code miss some loggings ,but it's easy to add.
check wandb examples https://docs.wandb.ai/guides/integrations/accelerate

I add a log of traning loss like:


```
if accelerator.is_main_process:
  print("Epoch {}, step {}, data_time: {}, time: {}, step_loss: {}".format(
                        epoch, step, load_data_time, time.perf_counter() - begin, avg_loss))
  # Log to wandb by calling `accelerator.log`, `step` is optional
  accelerator.log({"train_loss": avg_loss,}, step=global_step)


```

**Comment by waldiwunder** [comment]
Thank you! You answer helped me a lot to get it running for tensorboard. Here the detailed description of what I did for anyone who encounters the same problem:

1. Install tensorboard with pip3 install tensorboard==2.14.0
2. Add to tutorial_train.py:

```python
accelerator_project_config = ProjectConfiguration(project_dir=args.output_dir, logging_dir=logging_dir)
accelerator = Accelerator(
        mixed_precision=args.mixed_precision,
        log_with=["tensorboard"],
        project_config=accelerator_project_config,
        project_dir=args.output_dir
)
if accelerator.is_main_process:
       project_name = datetime.datetime.now().strftime("%Y%m%dT%H%M%S")
        if args.output_dir is not None:
            os.makedirs(args.output_dir, exist_ok=True)
        accelerator.init_trackers(project_name=project_name)
        tracker = accelerator.get_tracker("tensorboard")
        print("tracker.logging_dir ", tracker.logging_dir)
``` 


And in training run:

```python
 if accelerator.is_main_process:
                        print("Epoch {}, step {}, data_time: {}, time: {}, step_loss: {}".format(
                        epoch, step, load_data_time, time.perf_counter() - begin, avg_loss))
                    accelerator.log({"train_loss": avg_loss, "loss": loss.item()}, step=global_step)
            
            global_step += 1
            
            if global_step % args.save_steps == 0:
                save_path = os.path.join(args.output_dir, f"checkpoint-{global_step}")
                accelerator.save_state(save_path, safe_serialization=False)
            begin = time.perf_counter()
accelerator.end_training()
``` 
3. python -m pip install protobuf==4.25.3 (with version 5.27.3 the plots are not shown in tensorboard)
4. Open event file with tensorboard --logdir .
